### ISSUE TITLE
feat: add gradient clipping to iPEPS AD optimization

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#194]
## Problem

iPEPS AD optimization (`optimize_gs_ad`) has gradient instability. Gradients through the CTM fixed-point implicit differentiation have occasional large spikes that cause the optimizer to overshoot and diverge.

Evidence from learning rate sweep (D=2, chi=16, SU init):
- lr=1e-3: stable, slowly improving (E: -0.549 → -0.551 in 20 steps)
- lr=5e-3: oscillating, gradient spike of 4.4 at step 5
- lr≥1e-2: diverges completely

The benchmark test `test_2site_heisenberg_ad_energy_benchmark` fails intermittently because AD doesn't reliably improve on SU initialization.

## Proposed fix

Add gradient clipping to `_optimize_gs_ad_tensor` and `_optimize_gs_ad_tensor_2site`:

```python
# After computing grads, clip gradient norm
grad_norm = jnp.sqrt(sum(jnp.sum(g**2) for g in jax.tree.leaves(grads)))
max_grad_norm = 1.0  # configurable via iPEPSConfig
grads = jax.tree.map(lambda g: g * jnp.minimum(1.0, max_grad_norm / (grad_norm + 1e-10)), grads)
```

Add `gs_max_grad_norm: float = 1.0` to `iPEPSConfig`.

## References
- Learning rate sweep results in this conversation
- TeNPy and peps-torch both use gradient clipping for iPEPS AD
